### PR TITLE
feat(spanner/spansql): support table_hint_expr at from_clause on query_statement

### PR DESCRIPTION
**Pull request by nakatamixi**
This fixes parse error when query statement includes table hint expr.
This add function `parseHints` and use in parsing table hints and join hints.

### CODE REVIEW

**Comment by hengfengli** [comment]
@olavloite Can you please review this PR? Thanks.

File: spanner/spansql/parser.go
Comment (olavloite) [inline spanner/spansql/parser.go:2115]: Hmm... It seems that the [documentation](https://cloud.google.com/spanner/docs/query-syntax#from_clause) is slightly off here. It is actually possible to have more than one table hint, so the variable here should be renamed to `hints` and the `parseHints(..)` function should be called with `true` rather than `false`.

For example the following query is accepted by Cloud Spanner:

```sql
select *
from albums@{FORCE_INDEX=_BASE_TABLE, GROUPBY_SCAN_OPTIMIZATION=FALSE}
```

File: spanner/spansql/parser.go
Comment (olavloite) [inline spanner/spansql/parser.go:2119]: nit: same as above, rename to `sf.Hints`

File: spanner/spansql/parser_test.go
Comment (olavloite) [inline spanner/spansql/parser_test.go:118]: nit: add an additional test that verifies that the parser also accepts multiple table hints.

File: spanner/spansql/sql.go
Comment (olavloite) [inline spanner/spansql/sql.go:371]: Same as above, it seems that the documentation is not completely clear on this, but Cloud Spanner does accept more than one table hint.

File: spanner/spansql/types.go
Comment (olavloite) [inline spanner/spansql/types.go:397]: nit: rename to `Hints`

**Review by olavloite — COMMENTED** [review]
Thanks for your contribution. This looks mostly good to me, but it seems that the documentation is not 100% correct on whether multiple table hints are allowed or not. Cloud Spanner actually does support more than one table hint, so we should try to implement it in the same way in this emulator. See my comments and example below.

**Comment by nakatamixi** [comment]
Thanks for your kind review.
I checked multiple table hint keys can use at my spanner instance.
And I fixed to `parseHints` always parse multiple keys.

**Review by olavloite — APPROVED** [review]
Thanks, LGTM

**Comment by olavloite** [comment]
@hengfengli I think we need an approval from either you or @skuruppu in order to merge this.

**Review by hengfengli — APPROVED** [review]

File: spanner/spansql/sql.go
Comment (hengfengli) [inline spanner/spansql/sql.go:374]: Can you please update to `i++`? I got an error from the linter: 

```
spanner/spansql/sql.go:374:4: should replace i += 1 with i++
```

Comment (nakatamixi) [reply spanner/spansql/sql.go]: Thanks.

fixed at 83d63c5

**Comment by hengfengli** [comment]
The integration test fails due to #4450. I will merge this PR after it is merged.

**Comment by hengfengli** [comment]
@nakatamixi Thank you for your contribution!

**Comment by nakatamixi** [comment]
Thank you!

**Pull request merged**
